apps/els_core/src/els_provider.erl

```diff
@@ -2,254 +2,33 @@
 
 %% API
 -export([
-    handle_request/2,
-    start_link/0,
-    available_providers/0,
-    cancel_request/1,
-    cancel_request_by_uri/1
-]).
-
--behaviour(gen_server).
--export([
-    init/1,
-    handle_call/3,
-    handle_cast/2,
-    handle_info/2,
-    terminate/2
+    handle_request/2
 ]).
 
 %%==============================================================================
 %% Includes
 %%==============================================================================
--include_lib("kernel/include/logger.hrl").
+-include("els_core.hrl").
+
+-callback handle_request(provider_request()) -> provider_result().
 
--callback handle_request(request(), any()) ->
+-type provider() :: atom().
+-type provider_request() :: {atom() | binary(), map()}.
```

Comment:
When is the request type passed in as a `binary()`?

Reply:
Probably an artifact of the past, I can fix.